### ISSUE TITLE
how is dev_split generated

### ISSUE THREAD

**Issue by scheiblr** [deepset-ai/FARM#512]
Hey guys,

when there is no dev set your framework has the ability to generate a dev set portion based on the training set. Now, when I have a config file which contains multiple models, will this generated dev set be the same for each of the models in every iteration? As I saw, you're setting the random seed etc. globally for many libraries. It would be nice, if a shuffling would therefore lead to a comparable result (based on the seed).

As far as I experienced, `train_test_split` from `sklearn.model_selection` will always return the same set, if `random_state` is the same and as far as I saw, you are also using sklearn but I did not see, that you pass an integer into it. I think it would be more robust concerning reproducibility (according to [sklearn's docs](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html)), if an int was passed into that function (i.e. the random seed). Maybe I miss something important. It would be nice to me to get some light into that topic.

**Comment by Timoeller** [comment]
True that, a seed for splitting would be nice. Though the dev splitting is not done through sklearns functionality because of multiprocessing. If you want to include this feature feel free to open a PR so we can review.

A quick alternative would be: How about you split of the dev set yourself and supply it to the models if you want to have more control over the sampling technique? Would that solve your issue?

**Comment by scheiblr** [comment]
Farm is not using the exact same function I mentioned above (`train_test_split`), but [StratifiedKFold](https://github.com/deepset-ai/FARM/blob/master/farm/data_handler/data_silo.py#L891) and [KFold](https://github.com/deepset-ai/FARM/blob/master/farm/data_handler/data_silo.py#L894) from `sklearn` which according to their [documentation](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html) have the same parameter `random_state`. Thus, as far as I understood, it would be enough to push through the seed, which is an int, from the config to that very code line. Maybe you have a suggestion how this could be done in a nice way?

Concerning the quick alternative (which I use right now):
I think to get a more robust results, it would be beneficial to have the opportunity to run the experiments with several dev/train splits. Maybe a variable in the config would be required to be able to turn that behavior on and off, e.g. 'dev_split_reproducible'?

**Comment by Timoeller** [comment]
Sorry than you are talking about cross validation dev split and not a normal dev split. Getting a seed in there should not be a problem. If you want create a PR with this feature. Please make the seed optional when initializing the DataSiloforCrossVal.

**Comment by scheiblr** [comment]
Maybe I was looking on the wrong code lines. I got there, when I was looking for the dev-split code. Did I end up at the wrong lines? Where would the normal dev split take place?

**Comment by scheiblr** [comment]
And yes, cross validation would be what I'd like to do. is there already such a functionality included and if how can I run this from within a json setup experiment?

**Comment by Timoeller** [comment]
Did you find the code for Crossval in our examples? Closing this now for inactivity, feel free to reopen.
